shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -75,38 +82,51 @@ - (instancetype)initWithFrame:(CGRect)frame
   if (self = [super init]) {
     _frame = frame;
     _blurRadius = blurRadius;
-    BOOL isUIVisualEffectViewImplementationValid = NO;
-    for (UIView* view in visualEffectView.subviews) {
-      if ([view isKindOfClass:NSClassFromString(@"_UIVisualEffectBackdropView")]) {
-        for (NSObject* filter in view.layer.filters) {
-          if ([[filter valueForKey:@"name"] isEqual:@"gaussianBlur"] &&
-              [[filter valueForKey:@"inputRadius"] isKindOfClass:[NSNumber class]]) {
-            isUIVisualEffectViewImplementationValid = YES;
-            [filter setValue:@(_blurRadius) forKey:@"inputRadius"];
-            view.layer.filters = @[ filter ];
-
-            // Stop looping through other filters because the filter array is changed.
-            break;
-          }
-        }
-      } else if ([view isKindOfClass:NSClassFromString(@"_UIVisualEffectSubview")]) {
-        // Make `_UIVisualEffectSubview` transparanet so it does not add addtional color to the
-        // blurred PlatformView.
-        view.layer.backgroundColor = UIColor.clearColor.CGColor;
-      }
-    }
-    if (isUIVisualEffectViewImplementationValid) {
-      _backdropFilterView = [visualEffectView retain];
-      _backdropFilterView.frame = _frame;
-    } else {
+    [PlatformViewFilter prepareIfNecessary:visualEffectView];
+    if (IsUIVisualEffectViewImplementationValid == NO) {
       FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation to "
                          "access the gaussianBlur CAFilter.";
-      _backdropFilterView = nil;
+      [self release];
+      return nil;
     }
+    NSObject* gaussianBlurFilter = [[GaussianBlurFilter copy] autorelease];
+    FML_DCHECK(gaussianBlurFilter);
+    UIView* backdropView = visualEffectView.subviews[IndexOfBackdropView];
+    [gaussianBlurFilter setValue:@(_blurRadius) forKey:@"inputRadius"];
+    backdropView.layer.filters = @[ gaussianBlurFilter ];
+
+    UIView* visualEffectSubview = visualEffectView.subviews[IndexOfVisualEffectSubview];
+    visualEffectSubview.layer.backgroundColor = UIColor.clearColor.CGColor;
+
+    _backdropFilterView = [visualEffectView retain];
+    _backdropFilterView.frame = _frame;
   }
   return self;
 }
 
++ (void)prepareIfNecessary:(UIVisualEffectView*)visualEffectView {
+  if (GaussianBlurFilter) {
+    return;
+  }
+  NSUInteger index = 0;
+  for (UIView* view in visualEffectView.subviews) {
+    if ([view isKindOfClass:NSClassFromString(@"_UIVisualEffectBackdropView")]) {
+      IndexOfBackdropView = index;
+      for (NSObject* filter in view.layer.filters) {
+        if ([[filter valueForKey:@"name"] isEqual:@"gaussianBlur"] &&
+            [[filter valueForKey:@"inputRadius"] isKindOfClass:[NSNumber class]]) {
+          GaussianBlurFilter = filter;
+          IsUIVisualEffectViewImplementationValid = YES;
```

Comment:
Do you still need this? You can check if the index < 0 I think?

Reply:
I think you were somehow looking at an old version of the code.